predicators/approaches/sme_pg3_analogy_approach.py

```diff
@@ -1,97 +1,38 @@
+from predicators.approaches.pg3_analogy_approach import PG3AnalogyApproach
 from predicators.envs.base_env import BaseEnv
-from predicators.ground_truth_models import get_gt_nsrts, get_gt_options
 from predicators.settings import CFG
 from predicators.structs import NSRT, LDLRule, LiftedAtom, \
     LiftedDecisionList, Predicate, Type, Variable
 
 
-class InitializedPG3Approach(PG3Approach):
-    """Policy-guided planning for generalized policy generation (PG3) with
-    initialized policy."""
+class SMEPG3AnalogyApproach(PG3AnalogyApproach):
+    """Use SME for cross-domain policy learning in PG3."""
 
     @classmethod
     def get_name(cls) -> str:
-        return "initialized_pg3"
-
-    @staticmethod
-    def _get_policy_search_initial_ldls() -> List[LiftedDecisionList]:
-        # Create base and target envs.
-        base_env_name = CFG.pg3_init_base_env
-        target_env_name = CFG.env
-        base_env = get_or_create_env(base_env_name)
-        target_env = get_or_create_env(target_env_name)
-        base_options = get_gt_options(base_env_name)
-        target_options = get_gt_options(target_env_name)
-        base_nsrts = get_gt_nsrts(base_env.get_name(), base_env.predicates,
-                                  base_options)
-        target_nsrts = get_gt_nsrts(target_env.get_name(),
-                                    target_env.predicates, target_options)
-        # Initialize with initialized policy from file.
-        if CFG.pg3_init_policy is None:  # pragma: no cover
-            # By default, use policy from base domain.
-            save_path = utils.get_approach_save_path_str()
-            pg3_init_policy_file = f"{save_path}_None.ldl"
-        else:
-            pg3_init_policy_file = CFG.pg3_init_policy
-        # Can load from a pickled LDL or a plain text LDL.
-        _, file_extension = os.path.splitext(pg3_init_policy_file)
-        assert file_extension in (".ldl", ".txt")
-        if file_extension == ".ldl":
-            with open(pg3_init_policy_file, "rb") as fb:
-                base_policy = pkl.load(fb)
-        else:
-            with open(pg3_init_policy_file, "r", encoding="utf-8") as f:
-                base_policy_str = f.read()
-            base_policy = utils.parse_ldl_from_str(base_policy_str,
-                                                   base_env.types,
-                                                   base_env.predicates,
-                                                   base_nsrts)
+        return "sme_pg3"
+
+    def _induce_policies_by_analogy(
+            self, base_policy: LiftedDecisionList, base_env: BaseEnv,
+            target_env: BaseEnv, base_nsrts: Set[NSRT],
+            target_nsrts: Set[NSRT]) -> List[LiftedDecisionList]:
         # Determine analogical mappings between the current env and the
         # base env that the initialized policy originates from.
         analogies = _find_env_analogies(base_env, target_env, base_nsrts,
```

Comment:
I think this is good right now - the only thing is maybe in the future we want to move from a very well defined analogy object to a function? (but this can/probably will be in a different approach) - just making a note.

Reply:
that's a good thought yeah. leaving for the future sounds good